param/__init__.py

```diff
@@ -1940,13 +1970,42 @@ def _validate(self, val):
 
         for n in val:
             if not isinstance(n, dt_types):
-                raise ValueError("DateRange '%s' only takes datetime types: %s"%(self.name,val))
+                raise ValueError("DateTimeRange '%s' only takes datetime types: %s"%(self.name,val))
+
+        start, end = val
+        if not end >= start:
+           raise ValueError("DateTimeRange '%s': end datetime %s is before start datetime %s."%(self.name,val[1],val[0]))
+
+        # Calling super(DateTimeRange, self)._check(val) would also check
+        # values are numeric, which is redundant, so just call
+        # _checkBounds().
+        self._checkBounds(val)
+
+
+# Compatibility:
+DateRange = DateTimeRange
+
+
+class CalendarDateRange(Range):
+    """
+    A date range specified as (start_date, end_date).
+
+    Dates must be specified as datetime-like types (see
```

Comment:
Should this be "datetime" or "date"? The latter seems correct to me, but if the former is true, it's fine.